sql/src/main/java/io/crate/metadata/settings/CrateSettings.java

```diff
@@ -1,1760 +1,369 @@
+    private final Settings initialSettings;
+    private final Map<String, ReferenceImplementation> referenceImplementationMap;
 
-        @Override
-        public TimeValue defaultValue() {
-            return TimeValue.timeValueMinutes(10);
-        }
+    private Settings settings;
 
-        @Override
-        public Setting parent() {
-            return UDC;
-        }
+    @Inject
+    public CrateSettings(ClusterService clusterService) {
+        logger = Loggers.getLogger(this.getClass(), clusterService.getSettings());
+        initialSettings = Settings.builder()
+            .put(BUILT_IN_SETTINGS_DEFAULTS_MAP)
+            .put(clusterService.getSettings())
+            .build();
+        this.settings = initialSettings;
+        referenceImplementationMap = buildReferenceMap();
+        clusterService.add(this);
+    }
 
-        @Override
-        public boolean isRuntime() {
-            return false;
+    @Override
+    public void clusterChanged(ClusterChangedEvent event) {
+        try {
+            // nothing to do until we actually recover from the gateway or any other block indicates we need to disable persistency
+            if (event.state().blocks().disableStatePersistence() == false && event.metaDataChanged()) {
+                Settings incomingSetting = event.state().metaData().settings();
+                settings = Settings.builder().put(initialSettings).put(incomingSetting).build();
+            }
+        } catch (Exception ex) {
+            logger.warn("failed to apply cluster settings", ex);
         }
-    };
 
-    public static final TimeSetting UDC_INTERVAL = new TimeSetting() {
-        @Override
-        public String name() {
-            return "interval";
-        }
+    }
 
-        @Override
-        public TimeValue defaultValue() {
-            return TimeValue.timeValueHours(24);
-        }
+    Settings settings() {
+        return settings;
+    }
 
-        @Override
-        public Setting parent() {
-            return UDC;
-        }
+    public Map<String, ReferenceImplementation> referenceImplementationMap() {
+        return referenceImplementationMap;
+    }
 
-        @Override
-        public boolean isRuntime() {
-            return false;
+    private Map<String, ReferenceImplementation> buildReferenceMap() {
+        Map<String, ReferenceImplementation> referenceMap = new HashMap<>(BUILT_IN_SETTINGS.size());
+        for (CrateSetting crateSetting : BUILT_IN_SETTINGS) {
+            settingsToReferenceMap(referenceMap, crateSetting);
         }
-    };
-
-    public static final StringSetting UDC_URL = new StringSetting("url", null, true, "https://udc.crate.io", UDC);
+        return referenceMap;
+    }
 
-    public static final NestedSetting PSQL = new NestedSetting() {
-        @Override
-        public String name() {
-            return "psql";
-        }
+    private void settingsToReferenceMap(Map<String, ReferenceImplementation> referenceMap,
```

Comment:
maybe rename this to something like `buildReferenceTree` or indicate via comments the result structure